### PR TITLE
Update heroku cli

### PR DESCRIPTION
**Pull request by joepvd**

### CODE REVIEW

**Comment by cotsog** [comment]
Cross-referencing related issue: https://github.com/travis-pro/team-build-env/issues/62

File: lib/travis/build/appliances/update_heroku.rb [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+          sh.if '"$TRAVIS_DIST" == precise || "$TRAVIS_DIST" == trusty' do
+            sh.fold "update_heroku" do
+              sh.echo "Updating Heroku", ansi: :yellow
+              shell = <<~EOF
+              bash -c '
+                if which heroku &>/dev/null; then
+                  rm -rf /usr/local/heroku
+                  apt-get purge -y heroku-toolbelt heroku
+                  cd /usr/lib
+                  curl -sSL https://cli-assets.heroku.com/heroku-linux-x64.tar.xz | tar Jx
```
Comment (BanzaiMan) [inline lib/travis/build/appliances/update_heroku.rb]: `tar` on Precise can do `-J`?

**Comment by BanzaiMan** [comment]
Let's test this, and we can merge.

File: lib/travis/build/appliances/update_heroku.rb (hunk 1 above)
Comment (BanzaiMan) [reply lib/travis/build/appliances/update_heroku.rb]: It does. https://travis-ci.org/BanzaiMan/travis_production_test/builds/434258720#L132

**Comment by joepvd** [comment]
https://staging.travis-ci.org/joepvd/travis-staging-test/builds/728785

**Comment by BanzaiMan** [comment]
@joepvd Is this necessary for every single build?

**Comment by BanzaiMan** [comment]
https://github.com/travis-ci/travis-ci/issues/10167

**Comment by joepvd** [comment]
I would not know how to limit the builds where this is happening. We could scan the configuration for the word `heroku`, but that does not help for when that command is hidden away in a separate file (e.g. in a script deployment provider).

As a side note, installing Heroku via apt will get the old version that has a security vulnerability.

**Pull request merged**